contracts/Swap/SwapRouter.sol

```diff
@@ -0,0 +1,172 @@
+pragma solidity 0.8.17;
+
+import "@openzeppelin/contracts-upgradeable/access/OwnableUpgradeable.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "./interfaces/IPancakeSwapV2Router.sol";
+import "./interfaces/ISwapRouter.sol";
+import "./interfaces/IVtoken.sol";
+import "./interfaces/IWBnb.sol";
+
+/**
+ * @title Venus's Pancake Swap Integration Contract
+ * @notice This contracts allows users to swap a token for another one and supply/repay with the latter.
+ * @dev For all functions that do not swap native BNB, user must approve this contract with the amount, prior the calling the swap function.
+ * @author 0xlucian
+ */
+contract SwapRouter is OwnableUpgradeable, ISwapRouter {
+    using SafeERC20 for IERC20;
+
+    address private wBNBAddress;
+    address private swapRouterAddress;
+
+    function initialize(address wBNBAddress_, address swapRouterAddress_) public initializer {
+        require(wBNBAddress_ != address(0), "Swap: wBNB address invalid");
+        require(swapRouterAddress_ != address(0), "Swap: Pancake swap address invalid");
+        __Ownable_init();
+        wBNBAddress = wBNBAddress_;
+        swapRouterAddress = swapRouterAddress_;
+    }
+
+    // **************
+    // *** EVENTS ***
+    // **************
+
+    ///@notice This event is emitted whenever a successful supply on behalf of the user occurs
+    event SupplyOnBehalf(address indexed supplier, address indexed vTokenAddress, uint256 indexed amount);
+
+    ///@notice This event is emitted whenever a successful repay on behalf of the user occurs
+    event RepayOnBehalf(address indexed repayer, address indexed vTokenAddress, uint256 indexed amount);
+
+    // **************
+    // *** ERRORS ***
+    // **************
+
+    ///@notice Error indicating that suplying to a given market failed.
+    error SupplyError(address supplier, address vToken, uint256 errorCode);
+
+    ///@notice Error indicating that repaying to given market failed.
+    error RepayError(address repayer, address vToken, uint256 errorCode);
+
+    ///@notice Error indicating wBNB address passed is not the expected one.
+    error WrongAddress(address expectedAdddress, address passedAddress);
```

Comment:
the order of declaration of events should be before the `initialize` function to get more redability.